src/header-wrapper.jsx

```diff
@@ -0,0 +1,332 @@
+import React, { PureComponent, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { renderToStaticMarkup } from 'react-dom/server';
+import ExportButton from './export-button.jsx';
+
+/* TODO: the different render methods are very similar, split it into a few components
+  and try to get rid of some duplication */
+class HeaderWrapper extends PureComponent {
+  getBaseCSS () {
+    const {
+      vHeaderColorText,
+      vFontFamily,
+      vHeaderColorSchema,
+      vHeaderAlignText
+    } = this.props;
+
+    const baseCSS = {
+      color: vHeaderColorText,
+      fontFamily: vFontFamily,
+      backgroundColor: vHeaderColorSchema,
+      textAlign: vHeaderAlignText
+    };
+    return baseCSS;
+  }
+
+  renderSecondDimensionTitles () {
+    const {
+      vFontFamily,
+      vLetterSizeHeader,
+      vExportToExcel,
+      sufixCells,
+      vNumDims,
+      measure_count,
+      LabelsArray,
+      SecondHeader,
+      vSeparatorCols,
+      nSecond,
+      vLetterSize,
+      ExtraLabelsArray
+    } = this.props;
+
+    if (vNumDims === 2) {
+      if (measure_count > 1) {
+        const thStyle = {
+          ...this.getBaseCSS(),
+          cursor: 'default',
+          fontSize: (16 + vLetterSizeHeader) + 'px',
+          height: '80px',
+          width: '230px',
+          verticalAlign: 'middle'
+        };
+        return (
+          <Fragment key="second-dimension-titles">
+            <th
+              className="fdim-cells"
+              rowSpan="2"
+              padding-left="20px"
+              style={thStyle}
+            >
+              <ExportButton excelExport={vExportToExcel} />
+              {LabelsArray[0]}
+            </th>
+            {SecondHeader.map((header, index) => {
+              const emptyStyle = {
+                color: 'white',
+                fontFamily: vFontFamily,
+                fontSize: (13 + vLetterSizeHeader) + 'px'
+              };
+              const style = {
+                ...this.getBaseCSS(),
```

Comment:
saving baseCSS to a variable before mapping would be better, right?
so we don't creat a new variable and run the getBaseCSS() function for every item in the SecondHeader array